boule/sphere.py

```diff
@@ -208,6 +208,44 @@ def normal_gravity(self, latitude, height, si_units=False):
         # Convert gamma from SI to mGal
         return gamma * 1e5
 
+    def normal_gravitation(self, height, latitude=0, si_units=False):
```

Comment:
```suggestion
    def normal_gravitation(self, latitude, height, si_units=False):
```

Reply:
This should be almost identical to `normal_gravity` so that it's easier for users to switch between them if needed.

Reply:
If I understand this correctly you want the order of these two parameters the same, which I can get onboard with.

I would like to default the latitude to a value so that it is not required. It isn't used in the calculation at all, and may confuse users if it asks for a non-relevant parameter.

Reply:
So this is a bit of a conundrum:

1. For latitude to be optional, it has to come after `height` but that means that `Sphere.normal_gravitation` will be different from `Sphere.normal_gravity`, `Ellipsoid.normal_gravity`, and `Ellipsoid.normal_gravitation` (which is not implemented yet). 
2. For `Sphere.normal_gravitation` to be compatible with everything else, we need to require latitude even if we're not using it (but it could be set to `None` by the user).

Unfortunately, there is no good way to resolve both points simultaneously. In fact, this will be even worse when `TriaxialEllipsoid` has gravity calculations because it will require `longitude` as well. 

To me, 2. is the better option because if `Sphere` and `Ellipsoid` behave exactly the same, then using `boule.MOON` or `boule.WGS84` doesn't matter for user scripts or other libraries using Boule. So I could have a gravity processing function that takes a Boule realization and give it `MOON` with lunar data or `WGS84` with Earth data without the function needing to know the difference. We already do this with the coordinate conversions that don't touch `longitude` but require it anyway for compatibility with other libraries.

Initially, we were going to have everything be the `Ellipsoid` class. We only introduced `Sphere` because there are singularities in the `normal_gravity` of ellipsoids with zero flattening.

Reply:
The latest commit makes your recommended change.

Is the plan to add longitude to both the Sphere and Ellipsoid classes once we have a functioning triaxial Ellipsoid?

Reply:
Thanks, @dabiged. I hope this makes sense but I'm open to suggestions on alternative ways to resolve this.

> Is the plan to add longitude to both the Sphere and Ellipsoid classes once we have a functioning triaxial Ellipsoid?

I only realized this as I was writing the reply above so I think we might have to do that and break the API a bit. We can wait until TriaxialEllipsoid has gravity calculations, though.

Reply:
This is tricky. I like to keep the same method design on every ellipsoid class, so if I have a workflow for something on the Earth I can simply change it to Mars without getting weird errors because I'm passing an extra `latitude` argument to `normal_gravity`. The same for any triaxial ellipsoid complaining about a missing `longitude` argument.

But I know it's weird to have methods with arguments that they don't actually use. This may be not only confusing when using it, but could also create the misleading sensation that the normal gravity of the Moon varies on `latitude`.

What if we keep the best of both worlds by making our methods sufficiently smart through variable-length arguments:
```python
def normal_gravity(*coordinates, si_units=False):
    """
    """
    ...
```
This way the methods could take any number of coordinates (up to three, obviously) be sufficiently smart to know what they mean.
- If `len(coordinates) == 1` -> `height`
- If `len(coordinates) == 2` -> `latitude` and `height`
- If `len(coordinates) == 3` -> `longitude`, `latitude` and `height`

With this solution we don't need to break the API.
For obvious reasons, an `Ellipsoid` could only get 2 or 3 coordinates and should raise an error if `len(coordinates) == 1`.
The same for `TriaxialEllipsoid`: it should only take 3 coordinates, and in that case we can just use the `normal_gravity(longitude, latitude, height, ...)` syntax.

I'm brainstorming, so feel free to disagree with this idea or, even better, improve it.

Reply:
I have had a go at implementing something like this in (https://github.com/fatiando/boule/pull/73/commits/c0104c1db265783b68c01de9dd3d1bb3b89b4463.) I think we need to use **kwargs to capture the arguments as keywords as a `*coordinate` style does not specify which coordinate is which height/latittude/longitude. I think 
`  
def normal_gravity(longitude=12, latitude=24, height=36 )
`
is far cleaner and less ambiguous than:
`    def normal_gravity([12, 24, 36])`

I have also updated the docstring in the sphere class to show that you can now pass lat/long/height to normal_gravity, or just height. 

There needs to be more work on this to deal with weird arguments etc, but is this along the lines of what you were thinking @santisoler ?

Reply:
The trade-off there is that with this type of code tab-completion no longer works and users have to read the entire docstring to figure out the order of arguments. That's one of the things I dislike about matplotlib (they use this quite a bit). 

If we go for a single Ellipsoid class, then every method could take the full set of coordinates and be done with it. This is already what we do with the coordinate conversions and would make everything consistent. Then you know that methods of Boule always take longitude, latitude, height/radius, in that order.

Reply:
In reply to @dabiged. I would try to avoid passing coordinates as `kwargs`. Mainly because this is not how the rest of the Fatiando libraries handle passing coordiantes to functions or methods. I think that will bring some awkwardness.
For example, I can create a regular grid using Verde and then use Boule to compute the normal gravity on each point of the grid. Currently I can do that through:
```python
import verde as vd
import boule as bl

coordinates = vd.grid_coordinates(region=(..., ..., ..., ...), spacing=..., extra_coords=1e3)
ellipsoid = bl.WGS84
ellipsoid.normal_gravity(*coordinates[1:])
```
I think this is much simpler than specifying each coordinate with keys.

In reply to @leouieda. I think every option has its drawbacks. I'm a little bit worried of users getting confused about how the `normal_gravity` is computed if we are asking for arguments that aren't used, but you are right we are actually doing so on coordinate conversion (although I can say that a general coordinate conversions would need the entire set of coordinates, while in this case happens that `longitude` on both systems are the same).

If we really want to keep tab-completion, I would go with your proposal and ask for the whole set of coordinates. In fact the previous snippet I wrote would be much simpler:
```python
import verde as vd
import boule as bl

coordinates = vd.grid_coordinates(region=(..., ..., ..., ...), spacing=..., extra_coords=1e3)
ellipsoid = bl.WGS84
ellipsoid.normal_gravity(*coordinates)
```